packages/marketplace/src/components/ui/developer-edition-modal/developer-edition-modal.tsx

```diff
@@ -42,24 +72,34 @@ export const DeveloperEditionModal: React.FC<DeveloperEditionModalProps> = ({
     description: name,
   }))
 
-  const initialValues: FormValues = {
-    developerList: dropdownOptions.map(({ value }) => value),
-  }
+  const [isSuccess, setSuccess] = React.useState<boolean>(false)
+  const [selectedDeveloper, setSelectedDeveloper] = React.useState<DeveloperModel | undefined>()
+  const dispatch = useDispatch()
+  const loading = useSelector(selectCreateDeveloperSubscriptionLoading)
+  const hasError = useSelector(selectCreateDeveloperSubscriptionError)
+
+  let content
 
-  if (!visible) {
-    return null
+  if (isSuccess) {
```

Comment:
Pls use if return instead of if else if ...

Reply:
@tanphamhaiduong  I have tried to create a render function but it end with I have to add a lots of parameters for this function so I think it is unnecessary. 

```
export const renderModalContent = (
  isSuccess,
  hasError,
  selectedDeveloper,
  setSuccess,
  dispatch,
  afterClose,
  dropdownOptions,
  loading,
  developerLists,
  setSelectedDeveloper,
) => {
  if (isSuccess) {
    return (
      <SuccessContent developer={selectedDeveloper} afterClose={handleAfterClose(setSuccess, dispatch, afterClose)} />
    )
  }
  if (hasError) {
    return <ErrorContent afterClose={handleAfterClose(setSuccess, dispatch, afterClose)} />
  }
  return (
    <DeveloperEditionContent
      dropdownOptions={dropdownOptions}
      loading={loading}
      afterClose={handleAfterClose(setSuccess, dispatch, afterClose)}
      onFormSubmit={handleFormSubmit(developerLists, dispatch, handleOnCreated(setSelectedDeveloper, setSuccess))}
    />
  )
}

```